vizro-core/src/vizro/actions/export_data_class_action.py

```diff
@@ -0,0 +1,110 @@
+import importlib
+from typing import Any, Dict, List, Literal
+
+from dash import Output, State, ctx, dcc
+
+from vizro.actions._filter_action import _filter
+from vizro.actions.filter_interaction_action import filter_interaction
+from vizro.managers import model_manager
+from vizro.models.types import CapturedActionCallable
+
+
+class ExportDataClassAction(CapturedActionCallable):
+    def __init__(self, *args, **kwargs):
+        self._args = args
+        self._kwargs = kwargs
+        # Fake initialization - to let other actions see that this one exists.
+        super().__init__(*args, **kwargs)
+
+    def _post_init(self):
+        """Post initialization is called in the vm.Action build phase, and it is used to validate and calculate the
+        properties of the CapturedActionCallable. With this, we can validate the properties and raise errors before
+        the action is built. Also, "input"/"output"/"components" properties and "pure_function" can use these validated
+        and the calculated arguments.
+        """
+        self._page_id = model_manager._get_model_page_id(model_id=self._action_id)
+
+        # Validate and calculate "targets"
+        targets = self._kwargs.get("targets")
+        if targets:
```

Comment:
This feels like an improvement on the old function version because we now validate `targets` upfront rather than at runtime. Is that right?

Reply:
Yep, that's right! 😄